From: =?utf-8?q?S=C3=A9bastien_Villemot?= <sebastien@debian.org>
Date: Wed, 25 Aug 2021 00:00:00 +0000
Subject: Fix chill.test.sh in the Debian-specific context

chill.test.sh loads src/cold/chill.lisp, which itself loads
src/compiler/vop-existsp.lisp, but using the SYS logical pathname host.  This
fails at build time because debian/rules has initialized the SYS logical host
to /usr/share/sbcl-source/ in the Lisp image (and this will also break in the
autopkgtest because the sbcl-source package is not installed there).

This patch therefore switches back the SYS logical host to the current build
directory.
---
 tests/chill.test.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/chill.test.sh b/tests/chill.test.sh
index a9ff748..cbabb5a 100644
--- a/tests/chill.test.sh
+++ b/tests/chill.test.sh
@@ -3,7 +3,8 @@
 set -e
 # The function SB-C:LOCATION-NUMBER does not get defined for all backends,
 # and may get dropped by the tree shaker.
-run_sbcl --load ../src/cold/chill.lisp \
+run_sbcl --eval "(sb-ext:set-sbcl-source-location \"$SBCL_PWD/..\")" \
+  --load ../src/cold/chill.lisp \
   --eval '(assert (eq :external (nth-value 1 (find-symbol "LOCATION-NUMBER" "SB-C"))))' \
   --quit
 
